engine/src/main/java/org/hibernate/search/engine/environment/thread/spi/ThreadPoolProvider.java

```diff
@@ -42,9 +61,12 @@ public interface ThreadPoolProvider {
 	 * @param threads the number of threads
 	 * @param threadNamePrefix a label to identify the threads; useful for profiling.
 	 * @param queueSize the size of the queue to store Runnables when all threads are busy
+	 * @param batchingThreadContext a thread context that can be used to manipulate the execution environment, for example,
+	 * to initialize {@link ThreadLocal ThreadLocals}.
 	 * @return the new ExecutorService
 	 */
-	ThreadPoolExecutor newFixedThreadPool(int threads, String threadNamePrefix, int queueSize);
+	@Incubating
+	ThreadPoolExecutor newFixedThreadPool(int threads, String threadNamePrefix, int queueSize, BatchingThreadContext batchingThreadContext);
```

Comment:
I think you got this wrong; *this* method should have a default implementation. Otherwise you will break existing implementations of this SPI.